In [20]:
import pandas as pd
import numpy as np

In [21]:
df=pd.read_csv("C:\\Users\\hinaa\\Documents\\Python_DS_Pro\\Python_Analysis_Projects\\Covid_EDA\\covid_clinical_trails.csv")


In [22]:
df.shape

(5783, 27)

In [23]:
df.columns

Index(['Rank', 'NCT Number', 'Title', 'Acronym', 'Status', 'Study Results',
       'Conditions', 'Interventions', 'Outcome Measures',
       'Sponsor/Collaborators', 'Gender', 'Age', 'Phases', 'Enrollment',
       'Funded Bys', 'Study Type', 'Study Designs', 'Other IDs', 'Start Date',
       'Primary Completion Date', 'Completion Date', 'First Posted',
       'Results First Posted', 'Last Update Posted', 'Locations',
       'Study Documents', 'URL'],
      dtype='object')

In [24]:
df.head(10)

Rank   NCT Number                                              Title  \
0     1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   
1     2  NCT04595136  Study to Evaluate the Efficacy of COVID19-0001...   
2     3  NCT04395482  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3     4  NCT04416061  The Role of a Private Hospital in Hong Kong Am...   
4     5  NCT04395924         Maternal-foetal Transmission of SARS-Cov-2   
5     6  NCT04516954          Convalescent Plasma for COVID-19 Patients   
6     7  NCT04476940  COVID-19 Breastfeeding Guideline for African-A...   
7     8  NCT04634214  The Severity of COVID 19 in Diabetes and Non-d...   
8     9  NCT04602884  Early Detection of COVID-19 Using Breath Analysis   
9    10  NCT04384588  COVID19-Convalescent Plasma for Treating Patie...   

        Acronym                   Status         Study Results  \
0   COVID-IDNow   Active, not recruiting  No Results Available   
1      COVID-19       Not yet recruiting  No Results Available   
2   TAC-COVID19               Recruiting  No Results Available   
3      COVID-19   Active, not recruiting  No Results Available   
4  TMF-COVID-19               Recruiting  No Results Available   
5          CPCP  Enrolling by invitation  No Results Available   
6      COVID-BF       Not yet recruiting  No Results Available   
7       COVID19       Not yet recruiting  No Results Available   
8      COVID-19                Suspended  No Results Available   
9    FALP-COVID               Recruiting  No Results Available   

                                          Conditions  \
0                                            Covid19   
1                               SARS-CoV-2 Infection   
2                                            covid19   
3                                              COVID   
4  Maternal Fetal Infection Transmission|COVID-19...   
5                                           COVID 19   
6                    Covid19|Exclusive Breastfeeding   
7                             Covid19|Type2 Diabetes   
8                                            Covid19   
9  COVID-19 Infection|Cancer Patients|General Pop...   

                                       Interventions  \
0   Diagnostic Test: ID Now™ COVID-19 Screening Test   
1    Drug: Drug COVID19-0001-USR|Drug: normal saline   
2  Other: Lung CT scan analysis in COVID-19 patients   
3          Diagnostic Test: COVID 19 Diagnostic Test   
4  Diagnostic Test: Diagnosis of SARS-Cov2 by RT-...   
5           Biological: Convalescent COVID 19 Plasma   
6         Behavioral: COVID-19 Breastfeeding Support   
7                                                NaN   
8  Diagnostic Test: Breath biopsy sampling using ...   
9  Biological: Convalescent Plasma from COVID-19 ...   

                                    Outcome Measures  \
0  Evaluate the diagnostic performance of the ID ...   
1  Change on viral load results from baseline aft...   
2  A qualitative analysis of parenchymal lung dam...   
3  Proportion of asymptomatic subjects|Proportion...   
4  COVID-19 by positive PCR in cord blood and / o...   
5  Evaluate the safety|Change in requirement for ...   
6  COVID-19 breastfeeding guidance adherence at b...   
7  Severity of COVID 19 among people with and wit...   
8  Correlation between Volatile Organic Compounds...   
9  in-hospital mortality secondary to COVID-19 am...   

                               Sponsor/Collaborators  ...         Other IDs  \
0              Groupe Hospitalier Paris Saint Joseph  ...       COVID-IDNow   
1                         United Medical Specialties  ...  COVID19-0001-USR   
2                       University of Milano Bicocca  ...       TAC-COVID19   
3                    Hong Kong Sanatorium & Hospital  ...        RC-2020-08   
4  Centre Hospitalier Régional d'Orléans|Centre d...  ...      CHRO-2020-10   
5  Vinmec Research Institute of Stem Cell and Gen...  ...       ISC.20.11.1   
6                            Meharry Medical C

In [25]:
df.tail(10)

Rank   NCT Number                                              Title  \
5773  5774  NCT04639661  Predictors of Periodontal Outcomes Post-sanati...   
5774  5775  NCT04180709  CBT to Reduce Insomnia and Improve Social Reco...   
5775  5776  NCT04335643  Telehealth CBT for Adolescents and Young Adult...   
5776  5777  NCT04589377  Mindfulness to Mitigate Psychological Threat a...   
5777  5778  NCT04574466  Scaling-up Psychological Interventions With Sy...   
5778  5779  NCT04011644  Mobile Health for Alcohol Use Disorders in Cli...   
5779  5780  NCT04681339  Antibiotic Prescription in Children Hospitaliz...   
5780  5781  NCT04740229  Moderate-intensity Flow-based Yoga Effects on ...   
5781  5782  NCT04804917           3-year Follow-up of the Mind My Mind RCT   
5782  5783  NCT04680000  Chronic Pain Management In Primary Care Using ...   

           Acronym                   Status         Study Results  \
5773           NaN  Enrolling by invitation  No Results Available   
5774         CRISP               Recruiting  No Results Available   
5775          cSLE               Recruiting  No Results Available   
5776           NaN               Recruiting  No Results Available   
5777           NaN               Recruiting  No Results Available   
5778           NaN               Recruiting  No Results Available   
5779           NaN       Not yet recruiting  No Results Available   
5780           NaN               Recruiting  No Results Available   
5781  MindMyMindFU               Recruiting  No Results Available   
5782           NaN       Not yet recruiting  No Results Available   

                                             Conditions  \
5773            Periodontal Diseases|Periodontal Pocket   
5774                Psychotic Disorders|Psychosis|Sleep   
5775                       Systemic Lupus Erythematosus   
5776                                             Stress   
5777  Distress|PTSD|Anxiety|Depression|Trauma|Functi...   
5778                      Alcohol Drinking|Telemedicine   
5779  Community Acquired Pneumonia in Children|Antib...   
5780                               Stress|Psychological   
5781  Emotional Problem|Anxiety Disorder of Childhoo...   
5782                                       Chronic Pain   

                                          Interventions  \
5773                                                NaN   
5774                                    Device: Sleepio   
5775                                  Behavioral: TEACH   
5776                   Behavioral: Mindfulness Training   
5777                Behavioral: Problem Management Plus   
5778  Behavioral: A-CHESS self-monitored|Behavioral:...   
5779  Other: Antibiotic treatment|Other: No antibiot...   
5780                                   Behavioral: Yoga   
5781                                                NaN   
5782  Behavioral: Brief Cognitive Behavioral Therapy...   

                                       Outcome Measures  \
5773  Probing depth|Bleeding on probing|Tooth Loss|O...   
5774  Change from baseline Work and Social Adjustmen...   
5775  Recruitment rates of the study|Retention rates...   
5776  Mean differences in Psychological Threat measu...   
5777  Change in psychological distress|Change in pos...   
5778  Number of risky drinking days|Number of patien...   
5779  Antibiotic treatment rates in hospitalized chi...   
5780  Perceived Stress|Task switching paradigm|Digit...   
5781  The child's impact of mental health problems r...   
5782  Defense and Veterans Pain Rating Scale (DVPRS)...   

                                  Sponsor/Collaborators  ...  \
5773  Brock University|Dr. Peter C. Fritz, Periodont...  ...   
5774  University of Cambridge|Cambridgeshire and Pet...  ...   
5775  Michigan State University|Arthritis Foundation...  ...   
5776  University of Pittsburgh|U.S. National Science...  ...   
5777                               University of Zurich  ...   
5778  University of Wisconsin, Madison|National Inst.

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5783 entries, 0 to 5782
Data columns (total 27 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Rank                     5783 non-null   int64  
 1   NCT Number               5783 non-null   object 
 2   Title                    5783 non-null   object 
 3   Acronym                  2480 non-null   object 
 4   Status                   5783 non-null   object 
 5   Study Results            5783 non-null   object 
 6   Conditions               5783 non-null   object 
 7   Interventions            4897 non-null   object 
 8   Outcome Measures         5748 non-null   object 
 9   Sponsor/Collaborators    5783 non-null   object 
 10  Gender                   5773 non-null   object 
 11  Age                      5783 non-null   object 
 12  Phases                   3322 non-null   object 
 13  Enrollment               5749 non-null   float64
 14  Funded Bys              

In [27]:
df.describe()

Rank    Enrollment
count  5783.000000  5.749000e+03
mean   2892.000000  1.831949e+04
std    1669.552635  4.045437e+05
min       1.000000  0.000000e+00
25%    1446.500000  6.000000e+01
50%    2892.000000  1.700000e+02
75%    4337.500000  5.600000e+02
max    5783.000000  2.000000e+07

# Key_Insights

- The DataFrame contains 5783 entries and 27 columns.
- Data types include integer, float, and object (string).
- The 'Enrollment' column has some missing values.
- The 'Acronym', 'Interventions', 'Phases', 'Results First Posted', 'Study Documents', and 'Locations' columns have a considerable number of missing values.
- 'Rank' column ranges from 1 to 5783, with a mean of 2892.
- 'Enrollment' column ranges from 0 to 20,000,000, with a mean of approximately 18,320. The large standard deviation indicates high variability in enrollment across studies.
- Most studies have a small enrollment, as indicated by the median (50th percentile) value of 170.
- The 'Study Results' column indicates whether the study has reported results, but it's not clear what values it takes.
- The 'Status' column likely indicates the status of the study (e.g., recruiting, active, completed).
- 'Gender' column indicates the gender eligibility criteria for the studies.
- 'Age' column provides information about the age eligibility criteria for the studies.
- 'Sponsor/Collaborators' column lists the organizations sponsoring or collaborating on the studies.
- 'Study Type' likely describes the type of study (e.g., observational, interventional).
- 'Study Designs' column may contain information about the design of the studies (e.g., randomized control trial, cohort study).
- 'Start Date', 'Primary Completion Date', 'Completion Date', 'First Posted', and 'Last Update Posted' columns contain date information related to the studies.
- 'URL' column provides the URLs for accessing more information about the studies.

In [28]:
first_clinical_trial_data = df.iloc[0, :]

print(first_clinical_trial_data )

Rank                                                                       1
NCT Number                                                       NCT04785898
Title                      Diagnostic Performance of the ID Now™ COVID-19...
Acronym                                                          COVID-IDNow
Status                                                Active, not recruiting
Study Results                                           No Results Available
Conditions                                                           Covid19
Interventions               Diagnostic Test: ID Now™ COVID-19 Screening Test
Outcome Measures           Evaluate the diagnostic performance of the ID ...
Sponsor/Collaborators                  Groupe Hospitalier Paris Saint Joseph
Gender                                                                   All
Age                                18 Years and older   (Adult, Older Adult)
Phases                                                        Not Applicable

In [29]:
specific_trial_data = df.loc[df['NCT Number'] == 'NCT04595136', ['Title', 'Status']]
print(specific_trial_data)

                                               Title              Status
1  Study to Evaluate the Efficacy of COVID19-0001...  Not yet recruiting


In [30]:
recruiting_trials_data =df.loc[df['Status']=='Recruiting', ['Sponsor/Collaborators','Start Date']]
print(recruiting_trials_data )

                                  Sponsor/Collaborators         Start Date
2                          University of Milano Bicocca        May 7, 2020
4     Centre Hospitalier Régional d'Orléans|Centre d...        May 5, 2020
9     Fundacion Arturo Lopez Perez|Confederación de ...      April 7, 2020
10                                  The Christ Hospital     April 28, 2020
11                         University of Milano Bicocca     April 16, 2020
...                                                 ...                ...
5776  University of Pittsburgh|U.S. National Science...   October 26, 2020
5777                               University of Zurich    August 25, 2020
5778  University of Wisconsin, Madison|National Inst...     March 23, 2020
5780         University of Illinois at Urbana-Champaign  February 10, 2021
5781  Mental Health Services in the Capital Region, ...     March 22, 2021

[2805 rows x 2 columns]


In [31]:
first_five_df = df.loc[ :4, ['Title', 'Conditions', 'Outcome Measures']]
print(first_five_df)

                                               Title  \
0  Diagnostic Performance of the ID Now™ COVID-19...   
1  Study to Evaluate the Efficacy of COVID19-0001...   
2  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3  The Role of a Private Hospital in Hong Kong Am...   
4         Maternal-foetal Transmission of SARS-Cov-2   

                                          Conditions  \
0                                            Covid19   
1                               SARS-CoV-2 Infection   
2                                            covid19   
3                                              COVID   
4  Maternal Fetal Infection Transmission|COVID-19...   

                                    Outcome Measures  
0  Evaluate the diagnostic performance of the ID ...  
1  Change on viral load results from baseline aft...  
2  A qualitative analysis of parenchymal lung dam...  
3  Proportion of asymptomatic subjects|Proportion...  
4  COVID-19 by positive PCR in cord blood and / o..

In [32]:
last_three_trials_loc = df.loc[df.index[-3:], ['Completion Date', 'URL']]
print(last_three_trials_loc)

        Completion Date                                          URL
5780          July 2021  https://ClinicalTrials.gov/show/NCT04740229
5781  December 31, 2022  https://ClinicalTrials.gov/show/NCT04804917
5782      February 2025  https://ClinicalTrials.gov/show/NCT04680000


In [33]:
missing_values=df.isnull().sum()
print(missing_values)

Rank                          0
NCT Number                    0
Title                         0
Acronym                    3303
Status                        0
Study Results                 0
Conditions                    0
Interventions               886
Outcome Measures             35
Sponsor/Collaborators         0
Gender                       10
Age                           0
Phases                     2461
Enrollment                   34
Funded Bys                    0
Study Type                    0
Study Designs                35
Other IDs                     1
Start Date                   34
Primary Completion Date      36
Completion Date              36
First Posted                  0
Results First Posted       5747
Last Update Posted            0
Locations                   585
Study Documents            5601
URL                           0
dtype: int64


Missing Values

- Acronym                    3303
- Interventions               886
- Outcome Measures             35
- Gender                       10
- Phases                     2461
- Enrollment                   34
- Study Designs                35
- Other IDs                     1
- Start Date                   34
- Primary Completion Date      36
- Completion Date              36
- Results First Posted       5747
- Locations                   585
- Study Documents            5601


In [72]:
duplicates = df.duplicated()  
num_duplicates = duplicates.sum()

print('Sum of duplicate rows: ', num_duplicates)

Sum of duplicate rows:  0


In [73]:
enrollment_threshold = 200
threshold = df[ df['Enrollment'] > enrollment_threshold ]
print('Enrollment greater than a 200: ',len(threshold))


Enrollment greater than a 200:  2518


In [74]:
trails='No Results Available'

result=df[df['Study Results'] == trails]
print('Clinical trials have No Results Available: ',len(result))

Clinical trials have No Results Available:  5747


In [55]:
status1 = 'Completed'
status2 = 'Recruiting'
result = df[ (df['Status'] == status1) | (df['Status'] == status2) ]
num_trials = len(result)     
print(f"Number of trials in 'Completed' or 'Recruiting' status: {num_trials}")


Number of trials in 'Completed' or 'Recruiting' status: 3830


In [76]:
covid_trials = df[df['Acronym'] == 'COVID-19']
print('Clinical trials are related to COVID-19: ',len(covid_trials))

Clinical trials are related to COVID-19:  47


In [78]:
threshold_date = 'January 1, 2020'

after_2020 = df[df['Start Date'] > threshold_date]
print('Clinical trials started after January 1, 2020: ', len(after_2020))

Clinical trials started after January 1, 2020:  3674
